tests/integrations/logfire/test_utils.py

```diff
@@ -0,0 +1,506 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from pydantic import BaseModel, Field
+
+from mirascope.core.base._stream import BaseStream
+from mirascope.core.base._structured_stream import BaseStructuredStream
+from mirascope.core.base.call_response import BaseCallResponse
+from mirascope.core.base.metadata import Metadata
+from mirascope.core.base.tool import BaseTool
+from mirascope.integrations.logfire import _utils
+
+
+class FormatBook(BaseTool):
+    """Returns the title and author of a book nicely formatted."""
+
+    title: str = Field(..., description="The title of the book.")
+    author: str = Field(..., description="The author of the book in all caps.")
+
+    def call(self) -> str:
+        return f"{self.title} by {self.author}"  # pragma: no cover
+
+
+class MyCallResponse(BaseCallResponse):
+    @property
+    def content(self) -> str:
+        return "content"  # pragma: no cover
+
+    @property
+    def tools(self) -> list[BaseTool]:
+        return [FormatBook(title="The Name of the Wind", author="Rothfuss, Patrick")]
+
+
+patch.multiple(MyCallResponse, __abstractmethods__=set()).start()
+patch.multiple(BaseStream, __abstractmethods__=set()).start()
+
+
+class MyStream(BaseStream):
+    _provider = "test"
+
+    @property
+    def cost(self):
+        return 10
+
+
+@patch("logfire.with_settings", new_callable=MagicMock)
+def test_logfire_custom_context_manager(mock_logfire: MagicMock) -> None:
+    """Tests the `custom_context_manager` context manager."""
+    mock_fn = MagicMock()
+    mock_fn.__name__ = "mock_fn"
+    mock_fn.__annotations__ = {"metadata": Metadata(tags={"tag1", "tag2"})}
+
+    with _utils.custom_context_manager(mock_fn):
+        mock_logfire.assert_called_once()
+        call_args = mock_logfire.call_args[1]
+        assert call_args["custom_scope_suffix"] == "mirascope"
+        assert isinstance(call_args["tags"], list)
+        assert set(call_args["tags"]) == {"tag1", "tag2"}
+
+
+def test_get_call_response_span_data():
+    call_response = MyCallResponse(
+        metadata={"tags": {"version:0001"}},
+        response="hello world",
+        tool_types=[],
+        prompt_template="Recommend a {genre} book for me to read.",
+        fn_args={"genre": "nonfiction"},
+        dynamic_config={"computed_fields": {"genre": "nonfiction"}},
+        messages=[
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "text",
+                        "text": "Recommend a nonfiction book for me to read.",
+                    }
+                ],
+            }
+        ],
+        call_params={"tool_choice": "required"},
+        call_kwargs={
```

Comment:
why not just set `call_kwargs=MagicMock()` and `assert result["call_kwargs"] == call_response.call_kwargs`?

Seem unnecessary to have any real values here when we're just asserting that it's the same object, right?

Same is true for the other values here and in other tests.